CI: Cilium Cluster Mesh upgrade (ci-clustermesh): `(*ConnectivityTest).LogOwners` panic

## CI failure


test output:
```
📋 Test Report [cilium-test-1]
❌ 1/9 tests failed (1/187 actions), 105 tests skipped, 2 scenarios skipped:
Test [check-log-errors]:
  ❌ check-log-errors/no-errors-in-logs/kind-cluster2/kube-system/cilium-p8hhp (cilium-agent)
panic: runtime error: slice bounds out of range [2:1]

goroutine 1 [running]:
github.com/cilium/cilium/cilium-cli/connectivity/check.(*ConnectivityTest).LogOwners(0xc0008bf208, {0x7fed317c3a80, 0xc001270000})
	/go/src/github.com/cilium/cilium/cilium-cli/connectivity/check/logging.go:133 +0x67b
github.com/cilium/cilium/cilium-cli/connectivity/check.(*ConnectivityTest).report(0xc0008bf208)
	/go/src/github.com/cilium/cilium/cilium-cli/connectivity/check/context.go:514 +0xdb9
github.com/cilium/cilium/cilium-cli/connectivity/check.(*ConnectivityTest).PrintReport(0xc0010c27c8?, {0x62060b8?, 0xc000cf2200?})
	/go/src/github.com/cilium/cilium/cilium-cli/connectivity/check/context.go:469 +0x11b
github.com/cilium/cilium/cilium-cli/connectivity.Run({0x62060b8, 0xc000cf2200}, {0xc00012a340, 0x1, 0x1}, {0x7fed3182f7c0, 0x9583d20})
	/go/src/github.com/cilium/cilium/cilium-cli/connectivity/suite.go:46 +0x4ce
github.com/cilium/cilium/cilium-cli/cli.newCmdConnectivityTest.RunE.func1(0xc00035e908, {0x4502695?, 0x4?, 0x4502551?})
	/go/src/github.com/cilium/cilium/cilium-cli/cli/connectivity.go:106 +0x7b2
github.com/spf13/cobra.(*Command).execute(0xc00035e908, {0xc000791380, 0x1a, 0x1a})
	/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:985 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0xc000b8d508)
	/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(0x62161f0?)
	/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:1041 +0x13
main.main()
	/go/src/github.com/cilium/cilium/cilium-cli/cmd/cilium/main.go:22 +0xc5
``` 

https://github.com/cilium/cilium/actions/runs/13386077184/job/37383087410

--- a/cilium-cli/connectivity/check/logging.go
+++ b/cilium-cli/connectivity/check/logging.go
@@ -10,8 +10,8 @@ import (
 	"io"
 	"os"
 	"os/signal"
+	"regexp"
 	"runtime"
-	"strings"
 	"syscall"
 	"time"
 
@@ -113,6 +113,8 @@ func (s defaultScenario) Name() string {
 	return "cli-test-framework"
 }
 
+var ghWorkflowRegexp = regexp.MustCompile("^(?:.+?)/(?:.+?)/(.+?)@.*$")
+
 func (ct *ConnectivityTest) LogOwners(scenario ownedScenario) {
 	if !ct.params.LogCodeOwners {
 		return
@@ -126,11 +128,15 @@ func (ct *ConnectivityTest) LogOwners(scenario ownedScenario) {
 	}
 
 	var workflowOwners []codeowners.Owner
+	var ghWorkflow string
 	// Example: cilium/cilium/.github/workflows/conformance-kind-proxy-embedded.yaml@refs/pull/37593/merge
-	ghWorkflow := os.Getenv("GITHUB_WORKFLOW_REF")
-	ghWorkflow, _, _ = strings.Cut(ghWorkflow, "@")
-	segments := strings.Split(ghWorkflow, "/")
-	ghWorkflow = strings.Join(segments[2:], "/")
+	ghWorkflowRef := os.Getenv("GITHUB_WORKFLOW_REF")
+	matches := ghWorkflowRegexp.FindStringSubmatch(ghWorkflowRef)
+	// here matches should either be nil (no match) or a slice with two values:
+	// the full match and the capture.
+	if len(matches) == 2 {
+		ghWorkflow = matches[1]
+	}
 	if ghWorkflow != "" {
 		workflowRule, err := ct.CodeOwners.Match(ghWorkflow)
 		if err != nil || workflowRule == nil || workflowRule.Owners == nil {

